### PR TITLE
test(gmail): add unit tests for +triage argument parsing

### PR DESCRIPTION
**Pull request by anshul-garg27**
## Summary

- Add 8 unit tests for `gmail +triage` argument parsing — previously the only helper file with zero test coverage
- Tests cover: default/explicit/invalid `--max`, `--query` override, `--labels` flag, and `--format` selection

## Details

`src/helpers/gmail/triage.rs` had no `#[cfg(test)]` module. While the core logic involves network calls (Gmail API), the argument parsing and format selection are purely deterministic and worth covering.

## Tests added

- [x] `defaults_max_to_20_and_query_to_unread` — verifies default values
- [x] `explicit_max_overrides_default` — `--max 5` parses correctly
- [x] `non_numeric_max_falls_back_to_20` — `--max abc` silently falls back
- [x] `custom_query_overrides_default` — `--query 'from:boss'` works
- [x] `labels_flag_defaults_to_false` — default behavior
- [x] `labels_flag_set_when_passed` — `--labels` sets the flag
- [x] `format_defaults_to_table_when_absent` — table is the default
- [x] `format_json_when_specified` — `--format json` switches output

### CODE REVIEW

File: src/helpers/gmail/triage.rs [hunk 1]
```diff
@@ -173,3 +177,114 @@ pub async fn handle_triage(matches: &ArgMatches) -> Result<(), GwsError> {
...
+        let m = triage_cmd()
+            .try_get_matches_from(["triage", "--format", "json"])
+            .unwrap();
+        let fmt = m
+            .get_one::<String>("format")
+            .map(|s| crate::formatter::OutputFormat::from_str(s))
+            .unwrap_or(crate::formatter::OutputFormat::Table);
+        assert!(matches!(fmt, crate::formatter::OutputFormat::Json));
+    }
+}
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/helpers/gmail/triage.rs:290]: The test intentionally mirrors the parsing logic to verify the behavior — this is the same pattern used in `calendar.rs`, `chat.rs`, and `sheets.rs` tests which all build their own clap Command for testing. The `handle_triage` function requires network (Gmail API calls) so we can't call it directly in unit tests. The clap definition is the source of truth, and our test verifies the defaults match what the production code expects.
